amulet/level/formats/sponge_schem/format_wrapper.py

```diff
@@ -97,191 +151,117 @@ def _create(
-                numpy.array(packed_block_data, dtype=numpy.uint8)
-            )
-            if len(unpacked_block_data) != numpy.prod(size):
-                raise SpongeSchemReadError(
-                    "The data contained in BlockData does not match the size of the schematic."
-                )
-            dx, dy, dz = selection.shape
-            blocks_array: numpy.ndarray = numpy.transpose(
-                numpy.array(
-                    unpacked_block_data,
-                    dtype=numpy.uint32,
-                ).reshape((dy, dz, dx)),
-                (2, 0, 1),  # YZX => XYZ
-            )
-
-            if "Palette" not in sponge_schem:
-                raise SpongeSchemReadError(
-                    "Amulet is not able to read Sponge Schem files with no block palette."
-                )
-
-            palette_data = sponge_schem.get("Palette")
-            if not isinstance(palette_data, CompoundTag):
-                raise SpongeSchemReadError("Palette must be a CompoundTag.")
+        sponge_schem = load_nbt(f).compound
+        wrapper = sponge_schem.get("Schematic")
+        if isinstance(wrapper, CompoundTag):
+            sponge_schem = wrapper
 
-            block_palette: Dict[int, Block] = {}
-            for blockstate, index_tag in palette_data.items():
-                index = index_tag.py_int
-                if index in block_palette:
-                    raise SpongeSchemReadError(
-                        f"Duplicate block index {index} found in the palette."
-                    )
-                block_palette[index] = Block.from_string_blockstate(blockstate)
+        version = _detect_sponge_version(sponge_schem)
+        if version == 1:
+            raise SpongeSchemReadError("Unsupported Sponge schematic version 1")
 
-            if not numpy.all(numpy.isin(blocks_array, list(block_palette))):
+        offset_tag = sponge_schem.get("Offset")
+        if isinstance(offset_tag, IntArrayTag) and len(offset_tag) == 3:
+            min_point = numpy.array(offset_tag, dtype=numpy.int32)
+        else:
+            min_point = numpy.array([0, 0, 0], dtype=numpy.int32)
+
+        size = []
+        for key in ("Width", "Height", "Length"):
+            tag = sponge_schem.get(key)
+            if not isinstance(tag, ShortTag):
+                raise SpongeSchemReadError(f"Missing or invalid '{key}' tag")
+            size.append(tag.py_int & 0xFFFF)
+        max_point = min_point + size
+
+        selection = SelectionBox(min_point, max_point)
+        self._bounds[self.dimensions[0]] = SelectionGroup(selection)
+
+        dv = sponge_schem.get("DataVersion")
+        if not isinstance(dv, IntTag):
+            raise SpongeSchemReadError("Missing or invalid 'DataVersion' tag")
+        tv = self.translation_manager.get_version("java", dv.py_int)
+        self._platform, self._version = tv.platform, tv.data_version
+
+        if version == 2:
+            blk_ctn = sponge_schem
```

Comment:
This is redundant

Reply:
Fair enough, I'll strip that out and anything else, I see that feels/looks redundant.